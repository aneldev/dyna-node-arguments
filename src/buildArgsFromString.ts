export interface IDynaNodeArguments {
  root: string;

  [flag: string]: string;
}

export const buildArgsFromString = (cli: string): IDynaNodeArguments => {
  const output: IDynaNodeArguments = {
    root: '',
  };

  let flag = 'root';

  cli
    .split(' ')
    .filter(Boolean)
    .forEach(arg => {
      if (arg.indexOf('--') === 0) {
        flag = arg.substr(2) || 'root';
        if (!output[flag]) output[flag] = '';
        return;
      }
      output[flag] += `${output[flag] ? ' ' : ''}${arg.trim()}`;
    });

  return output;
};
