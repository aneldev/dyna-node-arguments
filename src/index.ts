import { buildArgsFromString } from "./buildArgsFromString";

export const dynaNodeArguments = buildArgsFromString(process.argv.slice(1).join(' '));
