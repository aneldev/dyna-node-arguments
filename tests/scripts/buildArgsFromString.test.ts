import "jest";

jasmine.DEFAULT_TIMEOUT_INTERVAL = 4000;

import { buildArgsFromString } from "../../src/buildArgsFromString";

describe('buildArgsFromString', () => {

  [
    'node build master.ts',
    'node build master.ts --ENV_MODE production',
    'node build master.ts --ENV_MODE production secure',
    'node build master.ts --ENV_MODE production secure --TARGET mobile android',
    'node build master.ts --ENV_MODE production secure -- basics --TARGET mobile android',
    'node build master.ts    --ENV_MODE    production     secure -- basics --TARGET mobile     android',
    'node build master.ts --ENV_MODE production -secure -- -verbose',
  ]
    .forEach(cli => {
      it(`Parse: "${cli}"`, () => {
        expect(buildArgsFromString(cli)).toMatchSnapshot();
      });
    });
});
