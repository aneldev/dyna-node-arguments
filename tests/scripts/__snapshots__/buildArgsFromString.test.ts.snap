// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildArgsFromString Parse: "node build master.ts    --ENV_MODE    production     secure -- basics --TARGET mobile     android" 1`] = `
Object {
  "ENV_MODE": "production secure",
  "TARGET": "mobile android",
  "root": "node build master.ts basics",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts --ENV_MODE production -secure -- -verbose" 1`] = `
Object {
  "ENV_MODE": "production -secure",
  "root": "node build master.ts -verbose",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts --ENV_MODE production secure -- basics --TARGET mobile android" 1`] = `
Object {
  "ENV_MODE": "production secure",
  "TARGET": "mobile android",
  "root": "node build master.ts basics",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts --ENV_MODE production secure --TARGET mobile android" 1`] = `
Object {
  "ENV_MODE": "production secure",
  "TARGET": "mobile android",
  "root": "node build master.ts",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts --ENV_MODE production secure" 1`] = `
Object {
  "ENV_MODE": "production secure",
  "root": "node build master.ts",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts --ENV_MODE production" 1`] = `
Object {
  "ENV_MODE": "production",
  "root": "node build master.ts",
}
`;

exports[`buildArgsFromString Parse: "node build master.ts" 1`] = `
Object {
  "root": "node build master.ts",
}
`;
